element_miniscope/miniscope.py

```diff
@@ -0,0 +1,849 @@
+
+            with open(recording_metadata.as_posix()) as f:
+                metadata = json.loads(f.read())
+
+            with open(recording_timestamps, newline= '') as f:
+                time_stamps = list(csv.reader(f, delimiter=','))
+
+            nchannels = 1 # Assumes a single channel
+            nframes = len(time_stamps)-1
+            px_height = metadata['ROI']['height']
+            px_width = metadata['ROI']['width']
+            fps = int(metadata['frameRate'].replace('FPS',''))
+            gain = metadata['gain']
+            spatial_downsample = 1 # Assumes no spatial downsampling
+            led_power = metadata['led0']
+            time_stamps = np.array([list(map(int, time_stamps[i]))
+                                       for i in range(1,len(time_stamps))])
+        else:
+            raise NotImplementedError(
+                f'Loading routine not implemented for {acquisition_software}'
+                ' acquisition software')
+
+        # Insert in RecordingInfo
+        self.insert1(dict(key,
+                          nchannels=nchannels,
+                          nframes=nframes,
+                          px_height=px_height,
+                          px_width=px_width,
+                          fps=fps,
+                          gain=gain,
+                          spatial_downsample=spatial_downsample,
+                          led_power=led_power,
+                          time_stamps=time_stamps))
+
+        # Insert file(s)
+        recording_files = [pathlib.Path(f).relative_to(
+                                            find_root_directory(
+                                                get_miniscope_root_data_dir(),
+                                                f)).as_posix() 
+                                for f in recording_filepaths]
+
+        self.File.insert([{**key, 
+                           'recording_file_id': i, 
+                           'recording_file_path': f} 
+                                for i, f in enumerate(recording_files)])
+
+
+# Trigger a processing routine -------------------------------------------------
+
+@schema
+class ProcessingMethod(dj.Lookup):
+    definition = """
+    # Method, package, analysis software used for processing of miniscope data 
+    # (e.g. CaImAn, etc.)
+    processing_method: char(16)
+    ---
+    processing_method_desc='': varchar(1000)
+    """
+
+    contents = [('caiman', 'caiman analysis suite')]
+
+
+@schema
+class ProcessingParamSet(dj.Lookup):
+    definition = """
+    # Parameter set used for processing of miniscope data
+    paramset_id:  smallint
+    ---
+    -> ProcessingMethod
+    paramset_desc: varchar(128)
+    param_set_hash: uuid
+    unique index (param_set_hash)
+    params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(cls, processing_method: str, paramset_id: int,
+                          paramset_desc: str, params: dict):
```

Comment:
I suggest optional insert of processing method in parent table. Suggestion below would also require an edit to the parent table insert

I think we also need a docstring here, especially if we're headed in the direction of documentation via site rendering of docstrings.

```suggestion
                          paramset_desc: str, params: dict,
                          processing_method_desc: str=''):
```